### PR TITLE
Remove common indentation of contents before yaml.load in page sandbox.

### PR DESCRIPTION
**Pull request by dzhuang**

### CODE REVIEW

File: course/sandbox.py [hunk 1]
```diff
@@ -344,4 +345,35 @@ def make_form(data=None):
...
+def remove_sb_content_common_indentation(code):
+    if "\n" not in code:
+        return code
+
+    lines = code.split("\n")
+
+    if lines:
+        base_indent = 0
+
+        # form field with textarea widget returns
```
Comment (dzhuang) [inline course/sandbox.py]: Maybe I am wrong in that the first line is stripped by codemirror.CodeMirrorTextarea?

File: course/sandbox.py [hunk 2]
```diff
@@ -344,4 +345,35 @@ def make_form(data=None):
...
+
+    lines = code.split("\n")
+
+    if lines:
+        base_indent = 0
+
+        # form field with textarea widget returns 
+        # cleaned_data with line[0] left stipped,
+        # so here we are starting from line[1].
+        while lines[1][base_indent] in " \t":
```
Comment (dzhuang) [inline course/sandbox.py]: It will fail when line[1] is " "?

**Comment by dzhuang** [comment]
The snippet is not OK, please don't merge.

**Comment by dzhuang** [comment]
The reason for #186 is my recent upgrade of Django from 1.8.x to 1.9.2, in which ``CharField`` add a new attribute ``strip`` with default value ``True`` [(link)]    (https://github.com/django/django/commit/11cac1bd8ef7546ca32d9969d4348bf412dc6664#diff-2645e84b6e5bc956ae62fe5f0a7d60faR372) .

BTW, it is surprising "whether or not strip by default for CharField content" had been discussed for almost 9 years!

**Comment by inducer** [comment]
Looks good. Thanks for tracking this down!

**Pull request merged**